tools/sklearn/pca.xml

```diff
@@ -0,0 +1,210 @@
+            #if $select_pca_type.select_solver_type.svd_solver_opts == 'arpack'
+                -tol $select_pca_type.select_solver_type.tolerance
+            #end if
+            $select_pca_type.whiten
+        #end if
+        #if $select_pca_type.select_pca_opts == 'incremental'
+            #if $select_pca_type.batch_size != ''
+                -b '$select_pca_type.batch_size'
+            #end if           
+            $select_pca_type.whiten
+        #end if
+        #if $select_pca_type.select_pca_opts == 'kernel'
+            -k '$select_pca_type.select_kernel_opts.kernel_opts'
+            #if $select_pca_type.select_kernel_opts.kernel_opts == 'poly'
+                #if $select_pca_type.select_kernel_opts.gamma != ''
+                    -g '$select_pca_type.select_kernel_opts.gamma'
+                #end if
+                -d '$select_pca_type.select_kernel_opts.degree'
+                -cf '$select_pca_type.select_kernel_opts.coef0'
+            #end if
+            #if $select_pca_type.select_kernel_opts.kernel_opts == 'rbf'
+                #if $select_pca_type.select_kernel_opts.gamma != ''
+                    -g '$select_pca_type.select_kernel_opts.gamma'
+                #end if
+            #end if
+            #if $select_pca_type.select_kernel_opts.kernel_opts == 'sigmoid'
+                #if $select_pca_type.select_kernel_opts.gamma != ''
+                    -g '$select_pca_type.select_kernel_opts.gamma'
+                #end if
+                -cf '$select_pca_type.select_kernel_opts.coef0'
+            #end if
+            -e '$select_pca_type.select_solver_type.eigen_solver_opts'
+            #if $select_pca_type.select_solver_type.eigen_solver_opts == 'arpack'
+                -tol $select_pca_type.select_solver_type.tolerance
+                #if $select_pca_type.select_solver_type.max_iter != ''
+                    -mi $select_pca_type.select_solver_type.max_iter
+                #end if
+            #end if
+        #end if
+        -o '$outfile'
+        -of '${outfile.ext}'
+        ]]>
+    </command>
+    <inputs>
+        <param name="infile" type="data" format="tabular" label="Input file"/>
+        <param name="header" type="boolean" label="Exclude Header" truevalue="--header" falsevalue="" help="If present, exclude the header row from the dataset"/>
+        <conditional name="select_cols">
+            <param name="select_cols_opts" type="select" label="Select Columns" help="Choose to select all columns, or exclude/include some">
+                <option value="all" selected="true">Select All</option>
+                <option value="include">Include</option>
+                <option value="exclude">Exclude</option>
+            </param>
+            <when value="include">
+                <param name="column_indices" type="text" value="0" label="Include Columns">
+                    <validator type="regex" message="Please specify only positive integers">(?:\d+\s)+\d+</validator>
```

Comment:
These validators seem complicated - it should be fine to just allow digits 0 to 9 and a space character, right? Something like: `<validator type="regex" message="Please specify only positive integers">^[0-9 ]+$</validator>`?